src/Interface/HasOrder.agda

```diff
@@ -2,31 +2,142 @@
 
 module Interface.HasOrder where
 
+open import Prelude hiding (_⊔_; suc; isPartialOrder; isEquivalence; trans)
 open import Level using (_⊔_; suc)
 open import Relation.Binary using (Rel)
-open import Relation.Binary.Definitions using (Decidable; Antisymmetric)
-open import Relation.Binary.Structures using (IsPreorder; IsPartialOrder)
+open import Relation.Binary.Definitions using (Decidable; Antisymmetric; Irreflexive; Asymmetric)
+open import Relation.Binary.Structures using (IsPreorder; IsPartialOrder; IsStrictPartialOrder; IsEquivalence)
+open import Relation.Binary.Structures using (IsDecTotalOrder ; IsStrictTotalOrder; IsTotalOrder)
 
-module _ {a ℓ ℓ₂} (A : Set a) (_≈_ : Rel A ℓ) where
+module _ {α ℓ} (A : Set α) (_≈_ : Rel A ℓ) where
 
-  record HasPreorder : Set (a ⊔ ℓ ⊔ suc ℓ₂) where
-    infix 4 _≤_
+  record HasPreorder : Set (α ⊔ suc ℓ) where
```

Comment:
I disagree with this idiom.  Lower case letters near the start of the alphabet are too important to waste on universe levels.  I decided to go with my preferred notation for universe levels here (alpha, beta, etc) because it was getting annoying to change things like `a = b` to `x = y`, to avoid conflicts with the universe level.

Reply:
Well, whether you disagree or not, our principal dependency `agda-stdlib` uses this convention everywhere, so you break consistency while gaining too little IMO. In this case, I presume you are referring to the implicit arguments `∀ {a b} → ...` which are confusing to begin with (typically one uses lowercase `a : A` and `b : B`, not `a b : A`).

Reply:
Please, let's not quibble about variable names. 🙏🏼

Reply:
You asked my review and, since I personally find such matters important, I communicated them. There's nothing wrong with ignoring some of them if you personally consider them insignificant, but I won't put special rules in place (i.e. to not mention matters involving variable names to you) depending on whose code I am reviewing (ego-less programming, etc..).